extensions/amp-minute-media-player/0.1/amp-minute-media-player.js

```diff
@@ -0,0 +1,480 @@
+  objOrParseJson,
+  originMatches,
+  redispatch,
+} from '../../../src/iframe-video';
+import {dev, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {
+  fullscreenEnter,
+  fullscreenExit,
+  isFullscreenElement,
+  removeElement,
+} from '../../../src/dom';
+import {getData, listen} from '../../../src/event-helper';
+import {installVideoManagerForDoc} from '../../../src/service/video-manager-impl';
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+/** @const */
+const TAG = 'amp-minute-media-player';
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpMinuteMediaPlayer extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    // Declare instance variables with type annotations.
+    /** @private {Element} */
+    this.iframe_ = null;
+
+    /** @private {string} */
+    this.contentType_ = null;
+
+    /** @private {?string} */
+    this.playerId_ = null;
+
+    /** @private {?string} */
+    this.contentId_ = '';
+
+    /** @private {?string} */
+    this.scannedElement_ = '';
+
+    /** @private {?string} */
+    this.tags_ = '';
+
+    /** @private {?string} */
+    this.minimumDateFactor_ = '';
+
+    /** @private {?string} */
+    this.scannedElementType_ = '';
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+
+    /** @private {?number} */
+    this.readyTimeout_ = null;
+  }
+
+  /**
+   * @param {boolean=} onLayout
+   * @override
+   */
+  preconnectCallback(onLayout) {
+    // Host that serves player configuration and content redirects
+    this.preconnect.url('https://www.oo-syringe.com', onLayout);
+  }
+
+  /** @override */
+  buildCallback() {
+    // Get attributes, assertions of values, assign instance variables.
+    // Build lightweight DOM and append to this.element.
+
+    const {element} = this;
+
+    this.contentType_ = userAssert(
+      element.getAttribute('data-content-type'),
+      'The data-content-type must be specified for <amp-minute-media-player> %s',
+      element
+    );
+
+    this.contentId_ =
+      this.contentType_ != 'semantic'
+        ? userAssert(
+            element.getAttribute('data-content-id'),
+            'The data-content-id must be specified for %s ' +
+              'data-content-type in <amp-minute-media-player> %s',
+            this.contentType_,
+            element
+          )
+        : '';
+
+    this.initSemanticFields_();
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+
+    // Warn if the player does not have video interface support
+    this.readyTimeout_ /*** NECESSARY?? ***/ /** @type {number} */ = Services.timerFor(
```

Comment:
What's the type issue that causes this coercion to be necessary? It doesn't seem to make sense, since this should be a promise?